packages/grafana-prometheus/src/components/PrometheusMetricsBrowser.tsx

```diff
@@ -347,9 +345,9 @@ export class UnthemedPrometheusMetricsBrowser extends React.Component<BrowserPro
   }
 
   componentDidMount() {
-    const { languageProvider, lastUsedLabels } = this.props;
+    const { languageProvider } = this.props;
     if (languageProvider) {
-      const selectedLabels: string[] = lastUsedLabels;
+      const selectedLabels: string[] = JSON.parse(localStorage.getItem(LAST_USED_LABELS_KEY) ?? `[]`) ?? [];
```

Comment:
To be safe, we should probably have this in `try-catch`. But also, have you considered wrapping `PrometheusMetricsBrowser` in `LocalStorageValueProvider` as it takes care of all the logic such as try catch/parsing/defaulting?

Reply:
This part of the code is changed in upcoming PR already: https://github.com/grafana/grafana/pull/101705
So I think we can skip try/catch here since that code will be replaced shortly.